public/js/charts/sensorCoverage.js

```diff
@@ -0,0 +1,113 @@
+// NOTE: requires('sheets');
+// NOTE: requires('aq');
+
+window.aq.charts.sensorCoverage = {};
+window.Chart.defaults.global.defaultFontSize = 10;
+const sensorsChart = document.getElementById("sensorsChart").getContext("2d");
+
+function getAllSensorTypes(arr) {
+    for (let i = 0; i < arr.length; i += 1) {
+      if (arr[i] instanceof Array) {
+        return getAllSensorTypes(arr[i]);
+      }
+      return Object.fromEntries(
+        Object.entries(arr[i]).filter(([key]) => key === 'sensordatavalues')
+      );
+    }
+    return null;
+  }
+
+async function handleLocationChange() {
+    const nodes = await window.aq.getNodes();
+    const countyCitiesMap = await window.sheets.getCountyCitiesMap();
+    const allCountyNodes = Object.entries(countyCitiesMap)
+        .map(([countyName, countyCities]) => {
+            const countyNodes = nodes.filter(({ location }) =>
+                countyCities.some(
+                    (city) =>
+                        location && location.city && location.city.indexOf(city) >= 0
+                )
+            );
+            return { name: countyName, value: countyNodes };
+        })
+        // Don't show counties with 0 nodes
+        .filter(({ value }) => value.length !== 0).map(value => value?.value);//Todo (Nyokabi) direct flatmap returns undefined???
+        const aLLNodesReduced = allCountyNodes[0].flatMap(node => node?.sensors).flatMap(node => node?.sensordatas).flatMap(node => node?.sensordatavalues).reduce(function(h, obj) {
+            h[obj.value_type] = (h[obj.value_type] || []).concat(obj);
+            return h; 
+          }, {});
+          const sensorTypes = Object.keys(aLLNodesReduced).map(key => {
+            return {
+                sensor: key, 
+                sensorAvg : Math.round(aLLNodesReduced[key].reduce((a, b) => a + (Number(b.value) || 0), 0)/aLLNodesReduced[key].length),
+            } 
+         });
+         const sensorCoverageData = sensorTypes?.filter(item  => item.sensor !== "timestamp" && item.sensor !== "height" && item.sensor !== "lon" && item.sensor !== "lat").map(item => item.sensorAvg);
```

Comment:
You can chain these calls

Reply:
@kilemensi Just pushed a fix on this. Could let me know if this works